src/System.Linq/src/System/Linq/Select.cs

```diff
@@ -625,13 +635,25 @@ public TResult TryGetLast(out bool found)
                 return sourceFound ? _selector(input) : default(TResult);
             }
 
+            private TResult[] LazyToArray()
```

Comment:
> It would probably be wise to avoid generating all that unnecessary code if it's never called.

Why is that the case for "case -1" but not the case for "default" in ToArray?  My point is that we should be consistent here.